BDArmory/Weapons/Missiles/MissileBase.cs

```diff
@@ -363,7 +367,9 @@ public void FloatingOriginCorrection()
                 {
                     if (craftPart.Current is null) continue;
                     if (craftPart.Current.name != missilePartName) continue;
-                    ++missilecount;
+                    var ammo = craftPart.Current.GetPart().FindModuleImplementing<ModuleMissileRearm>();
```

Comment:
When I was profiling stuff, this function gets called a lot in response to events during explosions (pretty much whenever a vessel loses a part), so we may want to optimise this ammo check. Something like
```csharp
ModuleMissileRearm ammo;
bool hasAmmo = true;
int AmmoCount // Returns the ammo count if the part contains ModuleMissileRearm, otherwise 1.
{
    get
    {
        if (!hasAmmo) return 1;
        if (ammo == null)
        {
            ammo = GetPart().FindModuleImplementing<ModuleMissileRearm>();
            if (ammo == null)
            {
                hasAmmo = false;
                return 1;
            }
        }
        return ammo.ammoCount;
    }
}
```
should work well. Then you can just use `missileCount += AmmoCount;` below.